protobuf/pg_query.proto

```diff
@@ -4108,3 +4108,44 @@ enum Token {
   MODE_PLPGSQL_ASSIGN3 = 777;
   UMINUS = 778;
 }
+
+
+message SummaryResult {
+  enum Context {
+    None = 0;
+    Select = 1;
+    DML = 2;
+    DDL = 3;
+    Call = 4;
+  }
+
+  message Table {
+    string table = 1;
+    Context context = 2;
+  }
+  // tables: Vec<(String, Context)>
+  repeated Table tables = 1;
+
+  // aliases: HashMap<String, String>
+  map<string, string> aliases = 2;
+
+  // cte_names: Vec<String>
+  repeated string cte_names = 3;
+
+  // functions: Vec<(String, Context)>
+  message Function {
+    string name = 1;
+    Context context = 2;
+  }
+  Function functions = 4;
+
+  // filter_columns: HashSet<(Option<String>, String)>
+  message FilterColumn {
+    // protobuf-c doesn't support optional fields, so I guess just... make
+    // table an empty string if it should be None-equivalent?
+    // Upstream issue: https://github.com/protobuf-c/protobuf-c/issues/476
```

Comment:
I think using empty strings here is the best solution - we'll have to do the same for some other fields (e.g. `Table.schema_name`)